session/session.go

```diff
@@ -0,0 +1,144 @@
+package session
```

Comment:
the only(mostly) reason we use packages to use them somewhere else, please do not duplicate the code base, use the pacage from the other PR.